packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmListQueryTests.kt

```diff
@@ -0,0 +1,250 @@
+        val result = querySample.objectListField.query("TRUEPREDICATE").max<Int>("intField").find()
+        assertEquals(5, result)
+    }
+
+    @Test
+    fun realmList_queryMin() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").min<Int>("intField").find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_queryCount() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").count().find()
+        assertEquals(6, result)
+    }
+
+    @Test
+    fun realmList_querySum() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").sum<Int>("intField").find()
+        assertEquals(15, result)
+    }
+
+    @Test
+    fun realmList_queryMaxEmptyList() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").max<Int>("intField").find()
+        assertEquals(null, result)
+    }
+
+    @Test
+    fun realmList_queryMinEmptyList() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").min<Int>("intField").find()
+        assertEquals(null, result)
+    }
+
+    @Test
+    fun realmList_queryCountEmptyList() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").count().find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_querySumEmptyList() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("FALSEPREDICATE").sum<Int>("intField").find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_queryMaxWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").max<Int>("nullableIntField").find()
+        assertEquals(4, result)
+    }
+
+    @Test
+    fun realmList_queryMinWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").min<Int>("nullableIntField").find()
+        assertEquals(0, result)
+    }
+
+    @Test
+    fun realmList_querySumWithNullValues() {
+        val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+        val result = querySample.objectListField.query("TRUEPREDICATE").sum<Int>("nullableIntField").find()
+        assertEquals(6, result)
+    }
+
```

Comment:
We also miss tests for the `IllegalArgumentException` being thrown if the query string is invalid or the argument input is wrong